### DISCUSSION TITLE
Error While Building iOS for the Flutter project.

### DISCUSSION THREAD

**Discussion by AdelDaniel** [codemagic-ci-cd/codemagic-docs#2811] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Just Running the workflow 

### Expected results

Build the iOS version

### Actual results

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/03-11-24_nafar4on.keychain-db
Searching for files matching /Users/builder/clone/chiton_buyer/ios/**/*.xcodeproj
Completed configuring code signing settings

> flutter build ipa --release --export-options-plist /Users/builder/export_options.plist --flavor production --target lib/main_production.dart

Archiving com.chitonco.chiton...
ios/Runner/AppDelegate.swift uses the deprecated @UIApplicationMain attribute, updating.
Automatically signing iOS for device deployment using specified development team in Xcode project: X8******FU
Running pod install...                                              3.8s
Running Xcode build...                                          
Xcode archive done.                                         11.3s
Failed to build iOS app
Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/FCResources.bundle

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.CountryCodes.json.QD3ajU

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.CountryListCell.nib.wI4092

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.CountryPickerController.nib.AmDfI3

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.FreshchatModels-Info.plist.MbzoFF

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.FreshchatSDK.v6jGF0

Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/FreshchatModel.momd

Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/FreshchatModels.bundle

Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/Headers

Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/Modules

Error (Xcode): Sandbox: rsync(2167) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/_CodeSignature

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.Info.plist.5QLggB

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.OBJCCountryView.nib.FaJJ0L

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.OBJCResendOTPView.nib.49vBqd

Error (Xcode): Sandbox: rsync(2169) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FreshchatSDK/FreshchatSDK.framework/.PrivacyInfo.xcprivacy.ndwmy4

Error (Xcode): Sandbox: rsync(2180) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/dSYMs

Error (Xcode): Sandbox: rsync(2180) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/Headers

Error (Xcode): Sandbox: rsync(2180) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/Modules

Error (Xcode): Sandbox: rsync(2180) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/_CodeSignature

Error (Xcode): Sandbox: rsync(2181) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/.FBSDKCoreKit_Basics.Vb7y14

Error (Xcode): Sandbox: rsync(2181) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/.Info.plist.2pgbJ3

Error (Xcode): Sandbox: rsync(2181) deny(1) file-write-create /Users/builder/Library/Developer/Xcode/DerivedData/Runner-chutkbvgtbbbbxcdavfduumdadvu/Build/Intermediates.noindex/ArchiveIntermediates/production/BuildProductsPath/Release-production-iphoneos/XCFrameworkIntermediates/FBSDKCoreKit_Basics/FBSDKCoreKit_Basics.framework/.PrivacyInfo.xcprivacy.nd33t9

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS

### Build id (optional)

67276f9123c254e23442b6e0

**Comment by MattyBoy4444** [comment]
There are 3 similar issues on Discord.  No answer from anyone yet.

**Comment by helinanever** [comment]
Hi everyone, we believe to have identified the root cause of the issue and have deployed a fix. Anyone affected, could you please run another build and confirm everything works as expected now?

**Comment by helinanever** [comment, marked as answer]
Here's some context about these errors. 

The issue was caused by the installation of `xcodeproj` version `1.26.0` on the build machines. This was done in an attempt to fix iOS code signing issues that projects created with Xcode 15.3 and newer were facing, you can read more about it here: https://github.com/orgs/codemagic-ci-cd/discussions/2787#discussioncomment-11073328

However, the changes introduced in `xcodeproj 1.26.0` turned out to break compatibility with CocoaPods for many projects, causing build failures.

We have now updated `xcodeproj` to `1.27.0` which [reverts the changes](https://github.com/CocoaPods/Xcodeproj/blob/master/CHANGELOG.md#1270-2024-10-31) made in the previous `xcodeproj` version and has been confirmed to resolve the issue.

If you run into problems with setting provisioning profiles in Codemagic, you can try reverting back to `1.26.0` by running
`gem uninstall xcodeproj --version 1.27.0`
before calling `xcode-project use-profiles`.

We're sorry for all the trouble this caused!
